### DISCUSSION TITLE
Color tokens missing from theme

### DISCUSSION THREAD

**Discussion by Pandroo** [material-components/material-web#5257] [Q&A] [answered]
I have generated my theme using the material-color-utilities library found here: https://www.npmjs.com/package/@material/material-color-utilities

This mostly works but the following tokens appear to be missing which makes certain components (filled buttons for example) to be off theme and have jarring default colours. Is there a way of implementing the theme generator to include the missing tokens or is this not implemented yet? If not, Is there a way of generating these colours based on my theme color in the mean time?

--md-sys-color-surface-bright
--md-sys-color-surface-dim
--md-sys-color-surface-container
--md-sys-color-surface-container-lowest
--md-sys-color-surface-container-low
--md-sys-color-surface-container-high
--md-sys-color-surface-container-highest

**Comment by asyncliz** [comment, marked as answer]
We suggest using the [material-web.dev](https://material-web.dev) color picker utility, which will generate a theme and allow you to copy all the correct tokens.

We're working on a better theme builder, but unfortunately right now the color utilities package uses an older token set schema.

You can also check out or copy our [catalog code](https://github.com/material-components/material-web/blob/main/catalog/src/utils/material-color-helpers.ts) which directly interacts with `@material/color-utilities` and adds the right token support. Reference that if you need to generate the theme dynamically at runtime.

**Reply by Pandroo** [reply]
Thank you! The catalog code worked perfectly
